diff --git a/base/Dockerfile b/base/Dockerfile
index 53e2fd7..f8f5a7f 100644
--- a/base/Dockerfile
+++ b/base/Dockerfile
@@ -3,13 +3,12 @@ ARG from
 FROM ${from}
 LABEL maintainer "NVIDIA CORPORATION <cudatools@nvidia.com>"
 
-RUN dpkg --add-architecture i386 && \
-    apt-get update && apt-get install -y --no-install-recommends \
-        libxau6 libxau6:i386 \
-        libxdmcp6 libxdmcp6:i386 \
-        libxcb1 libxcb1:i386 \
-        libxext6 libxext6:i386 \
-        libx11-6 libx11-6:i386 && \
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        libxau6 \
+        libxdmcp6 \
+        libxcb1 \
+        libxext6 \
+        libx11-6 && \
     rm -rf /var/lib/apt/lists/*
 
 # nvidia-container-runtime
diff --git a/glvnd/devel/Dockerfile b/glvnd/devel/Dockerfile
index f25cb5f..8882bb8 100644
--- a/glvnd/devel/Dockerfile
+++ b/glvnd/devel/Dockerfile
@@ -4,8 +4,8 @@ FROM ${from}
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
         pkg-config \
-        libglvnd-dev libglvnd-dev:i386 \
-        libgl1-mesa-dev libgl1-mesa-dev:i386 \
-        libegl1-mesa-dev libegl1-mesa-dev:i386 \
-        libgles2-mesa-dev libgles2-mesa-dev:i386 && \
+        libglvnd-dev \
+        libgl1-mesa-dev \
+        libegl1-mesa-dev \
+        libgles2-mesa-dev && \
     rm -rf /var/lib/apt/lists/*
diff --git a/glvnd/runtime/Dockerfile b/glvnd/runtime/Dockerfile
index 1bb328a..950b9f1 100644
--- a/glvnd/runtime/Dockerfile
+++ b/glvnd/runtime/Dockerfile
@@ -4,11 +4,11 @@ FROM ${from}
 LABEL maintainer "NVIDIA CORPORATION <cudatools@nvidia.com>"
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
-        libglvnd0 libglvnd0:i386 \
-        libgl1 libgl1:i386 \
-        libglx0 libglx0:i386 \
-        libegl1 libegl1:i386 \
-        libgles2 libgles2:i386 && \
+        libglvnd0 \
+        libgl1 \
+        libglx0 \
+        libegl1 \
+        libgles2 && \
     rm -rf /var/lib/apt/lists/*
 
 COPY 10_nvidia.json /usr/share/glvnd/egl_vendor.d/10_nvidia.json
