diff --git a/sanitizer_platform_limits_posix.cc.orig b/sanitizer_platform_limits_posix.cc
index b7fa6e8..2c42ece 100644
--- compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cc.orig
+++ compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cc
@@ -1128,7 +1128,7 @@ CHECK_SIZE_AND_OFFSET(ipc_perm, cuid);
 CHECK_SIZE_AND_OFFSET(ipc_perm, cgid);
 #if !defined(__aarch64__) || !SANITIZER_LINUX || __GLIBC_PREREQ (2, 21)
 /* On aarch64 glibc 2.20 and earlier provided incorrect mode field.  */
-CHECK_SIZE_AND_OFFSET(ipc_perm, mode);
+//CHECK_SIZE_AND_OFFSET(ipc_perm, mode);
 #endif
 
 CHECK_TYPE_SIZE(shmid_ds);
 