diff -Naur glew-1.13.0-orig/auto/core/gl/GL_OES_EGL_image glew-1.13.0/auto/core/gl/GL_OES_EGL_image
--- glew-1.13.0-orig/auto/core/gl/GL_OES_EGL_image	1969-12-31 16:00:00.000000000 -0800
+++ glew-1.13.0/auto/core/gl/GL_OES_EGL_image	2016-07-14 15:26:30.630873207 -0700
@@ -0,0 +1,6 @@
+GL_OES_EGL_image
+https://www.khronos.org/registry/gles/extensions/OES/OES_EGL_image.txt
+GL_OES_EGL_image
+	typedef void* GLeglImageOES
+	void glEGLImageTargetTexture2DOES (GLenum target, GLeglImageOES image)
+	void glEGLImageTargetRenderbufferStorageOES (GLenum target, GLeglImageOES image)
diff -Naur glew-1.13.0-orig/auto/Makefile glew-1.13.0/auto/Makefile
--- glew-1.13.0-orig/auto/Makefile	2015-08-10 04:47:22.000000000 -0700
+++ glew-1.13.0/auto/Makefile	2016-07-14 15:26:22.010947212 -0700
@@ -106,6 +106,7 @@
 	@echo "Downloading registry"
 	@echo "--------------------------------------------------------------------"
 	$(GIT_CLONE) registry
+	cd registry && git reset --hard 9438a0e6ea484bf12159323664cb8c4ff3091524
 	touch $@
 
 $(EXT)/.dummy: $(REGISTRY)/.dummy
